include/frozen/map.h

```diff
@@ -112,10 +114,10 @@ class map : private impl::CompareKey<Compare> {
       : map{items, Compare{}} {}
 
   /* element access */
-  constexpr Value const& at(Key const &key) const {
+  constexpr decltype(auto) at(Key const &key) const {
     return at_impl(*this, key);
   }
-  constexpr Value& at(Key const &key) {
+  constexpr decltype(auto) at(Key const &key) {
```

Comment:
This, unlike the explicit reference, allows our use of a proxy iterator by allowing this to return on-demand constructed by-value types to be returned here (mostly `string_view`/`frozen::string`).